### PR TITLE
The merge - add CODEOWNERS

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

Added CODEOWNERS 

## Links to any relevant issues

part of https://github.com/iotaledger/devx/issues/81

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+**/identity @iotaledger/identity
+
+# protocol - introduction-docs
+**/introduction-docs @iotaledger/shimmer-team
+
+# research
+**/iota-2.0-research-specifications @iotaledger/research
+
+# streams
+**/streams @iotaledger/streams
```
Comment (Dr-Electron) [inline .github/CODEOWNERS:37]: Do we even want to list this? We need to maintain it ourselves anyway until we finally remove it 😅

**Review by Dr-Electron — COMMENTED** [review]
LGTM, but why does it still contain errors? Is that fixable? Can we still merge?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# clients
+**/iota.rs @iotaledger/iota-sdk
+**/wallet.rs @iotaledger/iota-sdk
+**/wallet.rs @iotaledger/iota-sdk
```
Comment (Thoralf-M) [inline .github/CODEOWNERS:4]: Should one be `/iota-sdk`?

**Review by luca-moser — APPROVED** [review]

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+**/identity @iotaledger/identity
+
+# protocol - introduction-docs
+**/introduction-docs @iotaledger/shimmer-team
+
+# research
+**/iota-2.0-research-specifications @iotaledger/research
+
+# streams
+**/streams @iotaledger/streams
```
Comment (lucas-tortora) [inline .github/CODEOWNERS:37]:
```suggestion
```

**Comment by lucas-tortora** [comment]
> LGTM, but why does it still contain errors? Is that fixable? Can we still merge?

I'm not entirely sure. @luca-moser already gave the teams write access and made them public, but the error persists for some reason

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**